Application/Telegram/Middleware/Authentication.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Application.Features.UserFeature.Queries;
+using Domain.Enums;
+using MediatR;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Telegram.Bot.Types;
+
+namespace Application.Telegram.Middleware
+{
+    public static class Authentication
+    {
+        public static async Task<bool> Authenticate(Update update, IMediator _mediator)
+        {
+            var isAuthorized = UserStateStorage.userInfo.ContainsKey(update.Message.From.Id);
+            if (isAuthorized)
+                return true;
+
+            GetUserByIdQueryResponse user = null;
+            try
+            {
+                user = await _mediator.Send(new GetUserByIdQueryRequest { TelegramId = update.Message.From.Id });
+                UserRole role = (UserRole)Enum.Parse(typeof(UserRole), user.Roles.First());
+                UserStateStorage.AddUser(update.Message.From.Id, UserState.ProcessNotStarted, role);
```

Comment:
here you need this code as well, we need it for messages 
UserStateStorage.AddRecordToUserMessages(update.Message.From.Id, new List<int>());

Reply:
Ok